.github/workflows/osl_sync_main.yml

```diff
@@ -58,30 +67,36 @@ jobs:
           git merge --continue --no-edit
 
       - name: Run Bootstrap (No Conflicts)
-        if: "!contains(steps.merge.outputs.result, 'CONFLICT')"
+        if: env.MERGE_RESULT == 'SUCCESS'
         run: |
           pnpm bootstrap --no-frozen-lockfile
 
-      - name: Build (Rebuild Files)
-        run: >-
-          eval "pnpm ${{ env.PNPM_FILTER }} --workspace-concurrency=1 build:prod"
+      - name: Capture Uncommitted Changes
+        id: capture_changes
+        run: |
+          SYNC_CHANGES=$(git status --porcelain)
+          echo "SYNC_CHANGES=$SYNC_CHANGES" >> $GITHUB_ENV
+          if [ -n "$SYNC_CHANGES" ]; then
+            echo "HAS_CHANGES=true" >> $GITHUB_ENV
+          else
+            echo "HAS_CHANGES=false" >> $GITHUB_ENV
+          fi
 
       - name: Commit and Push Changes to Sync Branch
-        if: contains(steps.merge.outputs.result, 'CONFLICT') || git status --porcelain | grep -q '^[ACDM]'
+        if: env.MERGE_RESULT == 'CONFLICT' || env.HAS_CHANGES == 'true'
         run: |
           echo "Changes detected. Committing and pushing..."
-          SYNC_CHANGES=$(git status --porcelain) >> $GITHUB_ENV
+          echo "SYNC_CHANGES=${SYNC_CHANGES}"
```

Comment:
IIUC, this `SYNC_CHANGES` is just for the `osl_sync_pr_template.md` to use it, right? do we have to print it here with `echo`? Same question to line `99` (echo "SYNC_CHANGES=No conflicts or changes detected")

Reply:
I think it's nice to print it here so we have feedback in case of errors, no?

Reply:
ok